desktop-widgets/downloadfromdivecomputer.cpp

```diff
@@ -153,9 +153,15 @@ void DownloadFromDCWidget::DC##num##Clicked() \
 	ui.product->setCurrentIndex(ui.product->findText(qPrefDiveComputer::product##num())); \
 	bool isBluetoothDevice = isBluetoothAddress(qPrefDiveComputer::device##num()); \
 	bool isMacOs = QSysInfo::kernelType() == "darwin"; \
+	/* If we have a Bluetooth address, use it instead of rescanning. For \
+	MacOs, save the mount point.  This may be the wrong thing to do, but for \
+	the Catalina 10.15.7 laptop and Oceanic Atom 3.1 I use, the mountpoint \
+	is /dev/tty.usbserial-20030001 and it doesn't change regardless of the \
+	USB port being used.   If it isn't saved, it must be reentered. */ \
+	if (isBluetoothDevice || isMacOs) \
+		ui.device->setCurrentText(qPrefDiveComputer::device##num()); \
 	ui.bluetoothMode->setChecked(isBluetoothDevice); \
-	if (ui.device->currentIndex() == -1 || (isBluetoothDevice && !isMacOs)) \
-		/* macOS seems to have a problem connecting to remembered bluetooth devices  if it hasn't already had a connection in the current session */ \
+	if (ui.device->currentIndex() == -1 || isBluetoothDevice) \
 		ui.device->setCurrentIndex(deviceIndex(qPrefDiveComputer::device##num())); \
 	if (isMacOs) { \
```

Comment:
Not really required for this, but since you're already testing this with MacOS: Do you want to test if this extra code is still needed for MacOS, or if we can get a simplification by removing this - as stated, this is working fine without the extra code on other platforms.

Reply:
I removed this and don't see any adverse effects.    Given that I don't have a Windows or Linux platform to test on, it would be very helpful if someone could make sure I didn't break anything on Linux or Windows.

Reply:
Happy to test this, but since this code is not run (and potentially even optimised away during build time) on non-MacOS I am fairly confident that removing it won't break anything there.